### PR TITLE
Use setuptools to build Python bindings

### PR DESCRIPTION
**Pull request by daniel-larraz**
This PR replaces scikit-build with setuptools to build the Python bindings. Our setup with scikit-build didn't work on Windows, and used the FindPythonInterp and FindPythonLibs modules that were removed in CMake 3.27 (see policy [CMP0148](https://cmake.org/cmake/help/v3.27/policy/CMP0148.html)). This PR also uses pip to install the Python bindings using the Wheel format instead of using the deprecated call to `setup.py install --single-version-externally-managed`, which used the obsolete .egg format.

A follow-up PR will leverage this new setup to package the Python bindings for all the supported platforms and Python implementation versions using [cibuildwheel](https://cibuildwheel.readthedocs.io/). In this PR, only a minimal number of changes to the current packaging scripts have been applied to keep the packaging process working in the meantime.

### CODE REVIEW

File: src/api/python/setup.py.in [hunk 1]
```diff
@@ -1,35 +1,108 @@
...
+    if sys.platform == 'darwin':
+        import platform
+        mac_ver = os.environ.get("MACOSX_DEPLOYMENT_TARGET")
+        if mac_ver is None:
+          version_str, _, _ = platform.mac_ver()
+          major_minor = version_str.split(".")[:2]
+          mac_ver = ".".join(major_minor)
+          os.environ.setdefault("MACOSX_DEPLOYMENT_TARGET", mac_ver)
+        arch = "${MACOS_ARCH}" # Optionally set by CMake
+        arch = platform.machine() if arch == "" else arch
```
Comment (mpreiner) [inline src/api/python/setup.py.in:60]:
```suggestion
        arch = "${MACOS_ARCH}" # Optionally set by CMake
        if not arch:
            arch = platform.machine()
```

File: contrib/packaging_python/mk_clean_wheel.sh [hunk 2]
```diff
@@ -51,10 +48,7 @@ fi
 echo "Configuring"
 rm -rf build_wheel/
 
-# This new command like is supposed to ensure that we use the python 
-# version from the virtual environment that is activated. 
-# Once we remove scikit, this will not be necessary.
-./configure.sh $CONFIG --python-bindings --name=build_wheel -DPython_FIND_VIRTUALENV=ONLY -DPYTHON_LIBRARY=$VIRTUAL_ENV/lib -DPYTHON_INCLUDE_DIR=$PYINCLUDE
+./configure.sh $CONFIG --python-bindings --name=build_wheel -DPython_FIND_VIRTUALENV=ONLY
```
[the remark this answers is not in the document]
Comment (daniel-larraz) [reply contrib/packaging_python/mk_clean_wheel.sh]: Testing now. I think you are right, it shouldn't be necessary.

File: src/api/python/CMakeLists.txt [hunk 3]
```diff
@@ -129,81 +81,67 @@ add_custom_command(
...
+    ${CVC5_PYTHON_BASE_LIB}
   COMMAND
-    ${CMAKE_COMMAND} -E copy_directory
-    "${CVC5PythonicAPI_BASEDIR}/cvc5_pythonic_api"
-    "${CMAKE_CURRENT_BINARY_DIR}/cvc5/pythonic"
-  # Remove the license of the Pythonic API from the package's source files.
-  # The license file is included in the package's dist-info dir by setup()
+    # Force a new build if any dependency has changed
+    ${CMAKE_COMMAND} -E remove cvc5_python_base.cpp &&
+    ${CMAKE_COMMAND} -E remove cvc5/cvc5_python_base.${PYTHON_EXT}
```
[the remark this answers is not in the document]
Comment (daniel-larraz) [reply src/api/python/CMakeLists.txt:120]: The command `python setup.py build_ext --inplace` regenerates the C++ file and the cvc5_python_base.{so,pyd} file only if the cvc5_python_base.{so,pyd} file is not already present. Therefore, the rule removes these files if any dependencies have changed, ensuring that the setup.py script regenerates them as necessary.

Comment (mpreiner) [inline src/api/python/CMakeLists.txt:120]:
```suggestion
    ${CMAKE_COMMAND} -E remove cvc5_python_base.cpp cvc5/cvc5_python_base.${PYTHON_EXT}
```
let's avoid the && here if possible

**Review by mpreiner — APPROVED** [review]

**Pull request merged**